lib/dao.js

```diff
@@ -2687,6 +2687,14 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
   Model.notifyObserversOf('before save', context, function(err, ctx) {
     if (err) return cb(err);
+    if (Model.settings.forceId && ctx.instance[pkName] !== id) {
+      var err = new Error ('id property cannot be changed from ' +
+        id + ' to ' + inst[pkName] + ' in \'before save\' operation ' +
+        'hook when forceId is set to true');
```

Comment:
I don't have a strong opinion either. Here are factors to consider:

 - Backwards compatibility. How likely is it to have an existing app out there that was changing the `id` property in "before save" hook, and which would thus stop working if we start reporting an error?
 - Verbosity. If we print the warning for each hook invocation, the process output will get polluted very quickly.

Having wrote that, I think the ideal solution is to print a warning (that way we preserve backwards compatibility), but make sure to print it only once for each model (so that we don't pollute stdout/stderr).

Something along these lines (please rewrite it in a cleaner style):

```js
if (ctx.instance[pkName] !== id && !(Model._warned || {}).cannotOverwritePrimaryKeyInBeforeSaveHook) {
  Model._warned = Model._warned || {};
  Model._warned.cannotOverwritePrimaryKeyInBeforeSaveHook = true;
  console.warn(...);
}
```

Thoughts?